src/web/client/test/integration/urlBuilderUtil.test.ts

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import sinon from "sinon";
+import {
+    getParameterizedRequestUrlTemplate,
+    sanitizeURL,
+    updateEntityId,
+    pathHasEntityFolderName,
+    getRequestURL,
+    getCustomRequestURL,
+    getPatchRequestUrl,
+} from "../../utilities/urlBuilderUtil";
+import { expect } from "chai";
+import {
+    schemaKey,
+    schemaEntityKey,
+    schemaEntityName,
+    entityAttributesWithBase64Encoding,
+} from "../../schema/constants";
+import { httpMethod } from "../../common/constants";
+import * as schemaHelper from "../../utilities/schemaHelperUtil";
+
+import PowerPlatformExtensionContextManager, {
+    IPowerPlatformExtensionContext,
+} from "../../powerPlatformExtensionContext";
+describe("URLBuilder", () => {
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    it("getParameterizedRequestUrlTemplate_should_return_SINGLE_ENTITY_URL_KEY_when_isSingleEntity_is_true", async () => {
+        const powerPlatformExtensionContext: IPowerPlatformExtensionContext = {
+            dataSourcePropertiesMap: new Map<string, string>([
+                [schemaKey.SINGLE_ENTITY_URL, schemaKey.SINGLE_ENTITY_URL],
+            ]),
+        } as IPowerPlatformExtensionContext;
+        sinon
+            .stub(
+                PowerPlatformExtensionContextManager,
+                "getPowerPlatformExtensionContext"
+            )
+            .returns(powerPlatformExtensionContext);
+        const isSingleEntity = true;
+        const result = getParameterizedRequestUrlTemplate(isSingleEntity);
+        expect(result).eq("singleEntityURL");
```

Comment:
why not use constants value for this one like below?

Reply:
Thanks. I will update that in my next PR